assets/js/analyzer.js

```diff
@@ -1,5 +1,6 @@
 var analyzers = {}, analyzerData = {};
 var currentAnalyzerRequest;
+var THRESHOLD_REQUEST_LENGTH_FOR_ANALYZE = 2020;
```

Comment:
Sounds like a good approach overall. However, I would recommend against

> Depending upon the endpoint I supply as the parameter, I would handle the success accordingly!

Instead, pass in callbacks/extra options to the generalized function.